docs: update troubleshoot docs and add krew uses (#11017)
rook/rook: Documentation/Troubleshooting/krew-plugin.md

> ---

Reviewer: Let's actually keep this doc for now. It's ok for visibility to help them install it, I just don't want to duplicate more commands to this page, so they can refer to the krew docs for the commands.

Author: reverted.

    kubectl krew install rook-ceph
  ```

> ## Cluster Health

Reviewer: Honestly I'm not sure we want to put much effort into adding context to the krew plugin page in the Rook docs. It wasn't captured well in the discussion for https://github.com/rook/kubectl-rook-ceph/pull/58, but I thought we decided to just keep details of the Krew plugin commands in the Krew docs. The only place we would have in the Rook docs is where we integrate Krew commands on the other Documentation pages, like the troubleshooting guide. I don't think we want to maintain details on this main krew page in the Rook docs since it's really duplicative of the Krew docs.

Author: agree it is duplicate, removed.

> Health command check health of the cluster and common configuration issues. Health command currently validates these things configurations (let us know if you would like to add other validation in health command):

Reviewer 2: ```suggestion
Health command checks the health of the cluster and common configuration issues. Health command currently validates these things configurations (let us know if you would like to add other validation in the health command):
```

Author: ack

> 1. at least three mon pods should running on different nodes

Reviewer 2: ```suggestion
1. at least three mon pods should be running on different nodes
```

2. mon quorum and ceph health details

> 3. at least three osd pods should running on different nodes

Author: ack

4. all pods 'Running' status
5. placement group status
6. at least one mgr pod is running

Health commands logs have three ways of logging:

> 1. `Info`: This is just a logging information for the users.

Reviewer 2: ```suggestion
1. `Info`: This is just logging information for the users.
```

> 2. `Warning`: which mean there is some improvement required in the cluster.

Reviewer 2: ```suggestion
2. `Warning`: this means there is some improvement required in the cluster.
```

Author: ack

> 3. `Error`: This reuires immediate user attentions to get the cluster in healthy state.

Reviewer 2: ```suggestion
3. `Error`: This requires immediate user attention to get the cluster in a healthy state.
```

## Debug Mode

> Debug mode can be useful when a mon or OSD needs advanced maintenance operations that require the daemon to be stopped. Ceph tools such as `ceph-objectstore-tool`,`ceph-bluestore-tool`, or `ceph-monstore-tool` are commonly used in these scenarios. Debug mode will set up the mon or OSD so that these commands can be run.

Reviewer 2: ```suggestion
Debug mode can be useful when a mon or OSD needs advanced maintenance operations that require the daemon to be stopped. Ceph tools such as `ceph-objectstore-tool`, `ceph-bluestore-tool`, or `ceph-monstore-tool` are commonly used in these scenarios. Debug mode will set up the mon or OSD so that these commands can be run.
```
